### DISCUSSION TITLE
自定义函数中如何取到单元格内容

### DISCUSSION THREAD

**Discussion by lipengfei-6** [jxlsteam/jxls#269] [Q&A] [answered]
![image](https://github.com/jxlsteam/jxls/assets/125971393/8b2307d3-a66a-4154-bb1d-8d518c78b453)
如上情况：自定义函数要取合计的值，在自定义函数中应该以什么格式来写？${utils:amountConvert(V10)} 这样写的话会报错的

**Comment by SoltauFintel** [comment]
Please write us more about utils:amountConvert

**Reply by lipengfei-6** [reply]
utils：amountConvert是我自定义的JxlsUtils中的一个金额转换函数，并添加到

Map<String, Object> funcs = new HashMap<String, Object>();
            funcs.put("utils", new JxlsUtils());    //添加自定义功能
            jexlBuilder.namespaces(funcs);
            JexlEngine engine= new Engine(jexlBuilder);
            evaluator.setJexlEngine(engine);

**Reply by SoltauFintel** [reply, marked as answer]
I think the problem is the (cell?) "V10". AFAIK you can not use cell references inside ${...}

If you calculate in Excel use Excel functions. Otherwise calculate in Java.
